factcast-factus/src/main/java/org/factcast/factus/aggregate/cache/AbstractAggregateCache.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright © 2017-2025 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.factus.aggregate.cache;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.*;
+import java.util.*;
+import java.util.Set;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import lombok.*;
+import lombok.extern.slf4j.Slf4j;
+import org.factcast.core.*;
+import org.factcast.core.spec.*;
+import org.factcast.core.subscription.Subscription;
+import org.factcast.core.subscription.SubscriptionRequest;
+import org.factcast.core.subscription.SubscriptionRequestTO;
+import org.factcast.core.subscription.observer.FactObserver;
+import org.factcast.factus.Factus;
+import org.factcast.factus.projection.*;
+import org.factcast.factus.projector.FactSpecProvider;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.SmartInitializingSingleton;
+import org.springframework.core.GenericTypeResolver;
+
+@Slf4j
+@RequiredArgsConstructor
+public abstract class AbstractAggregateCache<A extends Aggregate>
+    implements AggregateCache<A>, DisposableBean, SmartInitializingSingleton {
+
+  private static final int DEFAULT_CACHE_SIZE = 100;
+  @Getter private Class<A> aggregateType;
+
+  protected final @NonNull Factus factus;
+  protected final @NonNull FactSpecProvider factSpecProvider;
+
+  protected final AtomicBoolean enabled = new AtomicBoolean(true);
+
+  protected Cache<@NonNull UUID, @NonNull A> cache;
+  protected Subscription subscription;
+
+  @Nullable
+  @Override
+  public A get(@NonNull UUID id) {
+    if (enabled.get()) {
+      return cache.getIfPresent(id);
+    } else return null;
+  }
+
+  @Override
+  public void put(@NonNull UUID id, @NonNull A aggregate) {
+    if (enabled.get()) cache.put(id, aggregate);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public void afterSingletonsInstantiated() {
+    aggregateType =
+        (Class<A>)
+            GenericTypeResolver.resolveTypeArgument(getClass(), AbstractAggregateCache.class);
+
+    cache =
+        configure(CacheBuilder.newBuilder())
+            // non negotiable:
+            .softValues()
```

Comment:
just for my understanding: what makes this necessary in combination with a maximum size?

Reply:
would go for 3. plus a friendly hint in the javadoc of `afterSingletonsInstantiated` saying to either leave it alone or to have the super call in the override